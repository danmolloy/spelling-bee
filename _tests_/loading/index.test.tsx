import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import LoadingGameIndex from "../../app/loading"

describe("<LoadingGameIndex />", () => {
  beforeEach(() => {
    render(<LoadingGameIndex />)
  })
  it("loading-index is in the document", () => {
    const loadingIndex = screen.getByTestId("loading-index")
    expect(loadingIndex).toBeInTheDocument()
  })
  it("matches snapshot", () => {

    const loadingIndex = screen.getByTestId("loading-index")
    expect(loadingIndex).toMatchSnapshot()
  })
})