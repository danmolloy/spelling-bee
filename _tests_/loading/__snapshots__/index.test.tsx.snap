// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingGameIndex /> matches snapshot 1`] = `
<div
  class="flex flex-col items-center"
  data-testid="loading-index"
>
  <div
    class=" md:p-4 flex flex-row md:justify-between w-full md:items-start justify-end "
    data-testid="header-div"
  >
    <div
      class="hidden md:flex flex-col"
    >
      <div
        class="flex flex-row items-end"
      >
        <h1
          class="font-display font-bold text-4xl pr-4"
        >
          Spelling Bee
        </h1>
        <div
          class=" animate-pulse h-8 w-60 bg-zinc-200 rounded-md m-1"
        />
      </div>
      <div>
        <h2>
          Originally a NYT game
        </h2>
        <h2>
          Replicated by Daniel Molloy
        </h2>
      </div>
    </div>
    <div
      class="flex flex-col items-end"
    >
      <div
        aria-label="Open/close menu"
        class="hover:bg-gray-100 active:bg-gray-200 text-2xl m-2 w-10 h-10 rounded-full flex items-center justify-center"
        data-testid="menu-icon"
        id="open-menu"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="flex flex-col md:flex-row-reverse w-full"
  >
    <div
      class="flex flex-col md:w-1/2 w-full md:px-2 items-center"
    >
      <div
        class="flex flex-row w-full items-center justify-center "
        data-testid="user-ranking"
      >
        <div
          class=" animate-pulse h-8 w-24 bg-zinc-200 rounded-md m-1"
        />
        <div
          class="w-full "
        >
          <div
            class=" flex flex-row justify-between items-center z-10 mr-2 "
          />
        </div>
      </div>
      <div
        class="w-full md:w-2/5 bg-white absolute z-30 border h-12 rounded-lg p-2 flex flex-row justify-between items-center font-light mt-10 transition-all duration-300"
        data-testid="word-list"
      >
        <div
          class=" animate-pulse h-4 w-full bg-zinc-200 rounded-md m-1"
          data-testid="loading-div"
        />
        <button
          aria-label="Show found words"
          class="w-8 h-8 flex justify-center items-center self-start rounded-full hover:bg-gray-100 active:bg-gray-200"
          data-testid="arrow-icon"
          id="show-found-words"
        >
          <div
            class="border-black border-r-2 border-b-2 h-3 w-3 self-end rotate-45 mb-3 "
          />
        </button>
      </div>
    </div>
    <div
      class="mt-16 flex flex-col items-center md:w-1/2"
      data-testid="input-index"
    >
      <div
        data-testid="text-input"
      >
        <h2
          class="font-bold text-2xl text-yellow-500 h-12 flex items-center justify-center  self-center"
        >
          <span
            class="flex text-yellow-300 text-3xl font-medium z-20 w-3 items-center justify-center mb-1 cursor"
          >
            |
          </span>
        </h2>
      </div>
      <div
        class="w-full fill-gray-300 justify-center flex flex-row md:p-6"
        data-testid="letters-div"
      >
        <div
          class="left flex flex-col h-full "
        >
          <button
            class="pulse hex active:scale-90 active:border-gray-50 my-16 rounded-full"
            data-testid="-hexagon-div"
            id="-letter"
            tabindex="0"
          >
            <div
              class="hex-top"
            />
            <div
              class="hex-middle flex flex-col justify-center items-center bg-zinc-200"
            />
            <div
              class="hex-bottom"
            />
          </button>
          <button
            class="pulse hex active:scale-90 active:border-gray-50 my-16 rounded-full"
            data-testid="-hexagon-div"
            id="-letter"
            tabindex="0"
          >
            <div
              class="hex-top"
            />
            <div
              class="hex-middle flex flex-col justify-center items-center bg-zinc-200"
            />
            <div
              class="hex-bottom"
            />
          </button>
        </div>
        <div
          class=" flex flex-col h-full -mt-10 md:-mt-14"
        >
          <button
            class="pulse hex active:scale-90 active:border-gray-50 my-16 rounded-full"
            data-testid="-hexagon-div"
            id="-letter"
            tabindex="0"
          >
            <div
              class="hex-top"
            />
            <div
              class="hex-middle flex flex-col justify-center items-center bg-zinc-200"
            />
            <div
              class="hex-bottom"
            />
          </button>
          <button
            class="pulse hex active:scale-90 active:border-gray-50 my-16 rounded-full"
            data-testid="-hexagon-div"
            id="-letter"
            tabindex="0"
          >
            <div
              class="center-top"
            />
            <div
              class="hex-center flex flex-col justify-center items-center bg-yellow-300"
            />
            <div
              class="center-bottom"
            />
          </button>
          <button
            class="pulse hex active:scale-90 active:border-gray-50 my-16 rounded-full"
            data-testid="-hexagon-div"
            id="-letter"
            tabindex="0"
          >
            <div
              class="hex-top"
            />
            <div
              class="hex-middle flex flex-col justify-center items-center bg-zinc-200"
            />
            <div
              class="hex-bottom"
            />
          </button>
        </div>
        <div
          class="flex flex-col"
        >
          <button
            class="pulse hex active:scale-90 active:border-gray-50 my-16 rounded-full"
            data-testid="-hexagon-div"
            id="-letter"
            tabindex="0"
          >
            <div
              class="hex-top"
            />
            <div
              class="hex-middle flex flex-col justify-center items-center bg-zinc-200"
            />
            <div
              class="hex-bottom"
            />
          </button>
          <button
            class="pulse hex active:scale-90 active:border-gray-50 my-16 rounded-full"
            data-testid="-hexagon-div"
            id="-letter"
            tabindex="0"
          >
            <div
              class="hex-top"
            />
            <div
              class="hex-middle flex flex-col justify-center items-center bg-zinc-200"
            />
            <div
              class="hex-bottom"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
